astropy/io/ascii/tests/test_ecsv.py

```diff
@@ -678,6 +705,31 @@ def test_read_bad_datatype():
         Table.read(txt, format='ascii.ecsv')
 
 
+def test_read_bad_datatype_v09():
+    """Test a malformed ECSV file for V0.9"""
+    txt = """\
+# %ECSV 0.9
+# ---
+# datatype:
+# - {name: a, datatype: object}
+# schema: astropy-2.0
+a
+fail
+[3,4]"""
+    try:
```

Comment:
I don't understand why you are catching an exception here. If this fails then it is a bona fide test failure and should be treated as such. Re-raising with an informative message is not typical in tests, normally that is reserved for helping out users. You can put that helpful text as just a comment or in the test docstring to help future devs if this ever fails.

Reply:
Good point, thanks. I've removed the unnecessary error catching from tests.